File: 3-automation-Playwright-Hoyts-Shop/pages/pageObjects/giftCardsPageHoytsShop.ts
import { Page, Locator } from "@playwright/test";
import { abstractComponentsHoytsShop } from "../abstractComponents/abstractComponentsHoytsShop";

export class giftCardsPageHoytsShop extends abstractComponentsHoytsShop {

    constructor(page: Page){
        super(page);
    }
    // ========================= Pick Gift Card =======================
    async pickGiftCard( title: string, value: string){
        const container = await this.scrollUntilProductVisible(title);
        await this.selectGiftCard(container, value);
    }
}

File: 3-automation-Selenium-Hoyts-Shop/src/main/java/hoytsShopPageObjects/giftCardsPageHoytsShop.java
package hoytsShopPageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import abstractComponents.hoytsShopAbstractComponents;

public class giftCardsPageHoytsShop extends hoytsShopAbstractComponents {

    public giftCardsPageHoytsShop(WebDriver driver) {
        super(driver);
    }

    
    // ========================= Action Methods =======================

    public void pickGiftCard(String title, String value) {
        WebElement container = scrollUntilProductVisible(title);
        selectGiftCard(container, value);
    }
}